src/wp-includes/block-bindings/sources/post-meta.php

```diff
@@ -0,0 +1,24 @@
+<?php
+/**
+ * Add the post_meta source to the Block Bindings API.
+ *
+ * @since 6.5.0
+ * @package WordPress
+ */
+function post_meta_source_callback( $source_attrs ) {
+	// Use the postId attribute if available
+	if ( isset( $source_attrs['postId'] ) ) {
+		$post_id = $source_attrs['postId'];
+	} else {
+		// $block_instance->context['postId'] is not available in the Image block.
+		$post_id = get_the_ID();
+	}
+
+	return get_post_meta( $post_id, $source_attrs['value'], true );
```

Comment:
Looking at this here I would love it if the function in core could already take care of looking up the post_id in the current context. 

I cannot think of any instance where I wouldn't want to get the ID of the current queried post within a query loop for example. So this code that is shown here will be needed in 100% of the cases where this API is used and feels error prone. If we could make it so that the callback is already getting passed the current post id :)

Reply:
Could we also pass the result through a filter here?

This could be useful for ACF to hydrate a raw post meta value into it's formatted value which users would expect.

Reply:
I don't think we should pass this through filters, this is its own abstractions if a third-party plugin wants its own kind of values, they can implement their own custom source.

Reply:
@fabiankaegy I'm not sure I understand the comment, can you clarify a bit?

Reply:
@youknowriad What I mean is that I would love it if the `wp_block_bindings_register_source` function already passed in the current post ID as a parameter of the callback function. 

Having to manually do the check to see if `$block_instance->context['postId']` is available and if so using it vs. accessing the id via the `get_the_ID` function is something that every single block binding source will need to do if it should work correctly in all context such as inside query loops. 

I would like it if the callback could be simplified to:

```php
php
 /**
  * Add the post_meta source to the Block Bindings API.
  *
  * @since 6.5.0
  * @package WordPress
  */
 function post_meta_source_callback( $source_attrs, $post_id ) {
 	// Override the current post id when an attribute called `postId` exists 
 	if ( isset( $source_attrs['postId'] ) ) {
 		$post_id = $source_attrs['postId'];
 	}

 	return get_post_meta( $post_id, $source_attrs['value'], true );
}
```

If I'm not mistaken there even is a bug in this implementation here. Because it doesn't use the context value at all. It should instead check for the availability of the context and if so use that over `get_the_ID`. And because of that very reason I think ideally core would already handle that check for context and pass the correct post id as a parameter to the callback.

Reply:
Personally it doesn't make sense to me. The sources can be anything, from posts to a random database table, to a call to some remote API... I don't think the signature should accommodate a specific source over any other.

Reply:
But don't sources always need to know which context they are rendered within?

Reply:
I’m still not sold on the name source, because it implies that the value is sourced from something external. In practice, the value used for replacement could be for example :
- current date
- translation
- hardcoded value

Overall, we are opening an API to dynamically compute the value of the attribute on the server so people don't need to default to custom and dynamic blocks.

Reply:
That context actually is helpful! In my mind, I was really mostly concerned with dynamic attributes on the post object / external API connections that need to be aware of the current context which is why I was so focussed on the Post ID. 

In that case an unrelated side note, I would love it if simply all blocks anywhere had the `usesContext: [ 'postId' ]` specified because the whole accessing the correct ID is an error-prone thing that often gets overlooked and the a feature doesn't work correctly in query loops etc